src/components/element-card/element-card.tsx

```diff
@@ -0,0 +1,101 @@
+import { CopyIcon } from "@lifesg/react-icons/copy";
+import { EnvelopeIcon } from "@lifesg/react-icons/envelope";
+import { EElementType, TElement } from "src/schemas";
+import { BaseCard, CardIcon } from "../common";
+import {
+    ActionButton,
+    ActionsContainer,
+    Container,
+    DetailsContainer,
+    IdLabel,
+} from "./element-card.styles";
+import { useBuilder } from "src/context-providers";
+
+interface IProps {
+    element: TElement;
+    onClick: () => void;
+    onDelete?: () => void;
+    onDuplicate?: () => void;
+}
+
+export const ElementCard = ({ element, onClick }: IProps) => {
+    // =========================================================================
+    // CONST, STATE, REFS
+    // =========================================================================
+    const { label, id } = element;
+    const { focusedElement } = useBuilder();
+
+    const isFocused = checkIsFocused();
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleDuplicateClick = (
+        event: React.MouseEvent<HTMLButtonElement>
+    ) => {
+        event.stopPropagation();
+        // TODO: Add handling
+    };
+
+    const handleDeleteClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        event.stopPropagation();
+        // TODO: Add handling
+    };
+
+    // =========================================================================
+    // HELPER FUNCTIONS
+    // =========================================================================
+    function checkIsFocused() {
+        return focusedElement.internalId === element.internalId;
+    }
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+    const renderIcon = () => {
+        let icon: React.ReactNode;
+
+        switch (element.type) {
+            case EElementType.EMAIL:
+                icon = <EnvelopeIcon />;
+                break;
+            default:
+                break;
+        }
+
+        return <CardIcon>{icon}</CardIcon>;
+    };
+
+    return (
+        <BaseCard onClick={onClick} focused={isFocused}>
+            <Container>
+                {renderIcon()}
+                <DetailsContainer>
+                    <Text.Body weight="semibold">{label}</Text.Body>
+                    <IdLabel weight="semibold">ID: {id}</IdLabel>
+                </DetailsContainer>
+                {isFocused && (
+                    <ActionsContainer>
+                        <ActionButton
+                            type="button"
+                            onClick={handleDuplicateClick}
+                        >
+                            <CopyIcon />
+                            Duplicate
+                        </ActionButton>
```

Comment:
- should have `$primary` by default
- need a disabled state

Reply:
From the looks of it, the duplicate was always greyed out. As for the disabled state, I can't find it in the Figma. Do u have a link?

Reply:
ok you can refer to this: https://www.figma.com/file/G9h7ux7jd3IGnJn6OEfDhC/%5BWeb%5D-Sequential?type=design&node-id=2963-22032&mode=design&t=OpLVEogw3w6KO5nU-0